frontstage/controllers/case_controller.py

```diff
@@ -0,0 +1,98 @@
+import logging
+
+import requests
+from flask import current_app as app
+from structlog import wrap_logger
+
+from frontstage.exceptions.exceptions import ApiError, InvalidCaseCategory
+
+
+logger = wrap_logger(logging.getLogger(__name__))
+
+
+def get_case_by_enrolment_code(enrolment_code):
+    logger.debug('Attempting to retrieve case by enrolment code', enrolment_code=enrolment_code)
+    url = f"{app.config['CASE_URL']}/cases/iac/{enrolment_code}"
+    response = requests.get(url, auth=app.config['CASE_AUTH'])
+
+    try:
+        response.raise_for_status()
+    except requests.exceptions.HTTPError:
+        log_level = logger.warning if response.status_code == 404 else logger.exception
+        log_level('Failed to retrieve case', enrolment_code=enrolment_code)
+        raise ApiError(response)
+
+    logger.debug('Successfully retrieved case by enrolment code', enrolment_code=enrolment_code)
+    return response.json()
+
+
+def get_case_by_party_id(party_id, case_events=False):
+    logger.debug('Attempting to retrieve case by party id', party_id=party_id)
```

Comment:
Does this not return a list of cases for the party id rather than just a singular case? This should be reflected in the logs.